OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -205,8 +205,8 @@ void ConnectionStateChanged(OrderManager om, string password, NetworkConnection
 					{
 						{ "initialMap", lastUpdatedUid ?? map.Uid },
 						{ "initialTab", MapClassification.System },
-						{ "onExit", Game.IsHost ? new Action(() => UpdateSelectedMap()) : null },
-						{ "onSelect", Game.IsHost ? onSelect : null },
+						{ "onExit", Game.IsHost ? (Action)UpdateSelectedMap : modData.MapCache.UpdateMaps },
+						{ "onSelect", Game.IsHost ? onSelect : _ => modData.MapCache.UpdateMaps() },
```

Comment:
Non-hosts cannot select a map, so this shouldn't be necessary? (Or am I missing something here?) And I assume this might break https://github.com/OpenRA/OpenRA/blob/b254eb0f3db83b5f592a3bb55713d3bd8077fa56/OpenRA.Mods.Common/Widgets/Logic/MapChooserLogic.cs#L55 ?

Reply:
Confirmed that the onSelect change regresses the Ok button, and that the onExit change fixes the crash.

Reply:
> Non-hosts cannot select a map, so this shouldn't be necessary? (Or am I missing something here?) And I assume this might break

You exit map chooser with either the Ok or Cancel buttons, they already update MapCache when you are host, so it also makes sense to update it for non-hosts

> Confirmed that the onSelect change regresses the Ok button, and that the onExit change fixes the crash.

Could you elaborate the regression? To me everything seems to be working as intended

Reply:
Ah I see. I didn't know the ok button was disabled. Fixed